db/migrations/3_claim_ticker_to_id.sql

```diff
@@ -0,0 +1,52 @@
+-- NOTE: This migration should only run after all pre 7.0 blocks have been processed
+-- Similar logic should be triggered when the 7.0 upgrade is reached
+
+-- Update claims ID field to use asset IDs
+WITH spec AS (
+    SELECT spec_version_id as version
+    FROM blocks
+    ORDER BY datetime DESC
+    LIMIT 1
+)
+UPDATE claims
+SET id =
+    REGEXP_REPLACE(id, '/[^/]+$', '/' || (scope->>'assetId')::TEXT),  -- Replace text after the last '/'
+    scope = jsonb_set(scope, '{value}', to_jsonb(scope->>'assetId'))
+FROM spec
+WHERE scope ? 'assetId' AND spec.version >= 7000000;
+
+-- Find ClaimRevoked events and update corresponding claims with the revoke date
+WITH spec AS (
+    SELECT spec_version_id as version
+    FROM blocks
+    ORDER BY datetime DESC
+    LIMIT 1
+),
+revoked_events AS (
+    SELECT
+           event_arg_1::json AS event_arg_json,
+           event_arg_0 || '/' ||
+           (SELECT json_object_keys(event_arg_1::json->'claim') LIMIT 1) || '/Asset/' ||
+           (event_arg_1::json->'claim'->(SELECT json_object_keys(event_arg_1::json->'claim') LIMIT 1)->>'Asset')
```

Comment:
It is not mandatory that claim revoked is always of type that has asset scope. What if CustomerDueDiligence or NoData type claim is revoked? I saw in staging data some of these instances to be there.

Reply:
The non scoped claims are correctly revoked with the current handler, so they don't need to be updated. Good catch with the other types though.